src/jit/instrxarch.h

```diff
@@ -0,0 +1,107 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef _INSTR_XARCH_H_
+#define _INSTR_XARCH_H_
+
+#if !defined(_TARGET_XARCH_)
+#error Unexpected target type
+#endif
+
+enum InsEncodingFmt : unsigned int
+{
+    // "R/M, [reg]" addressing mode
+    InsEncodingFmt_mr,
+
+    // "R/M, icon" addressing mode
+    InsEncodingFmt_mi,
+
+    // "reg, R/M" addression mode
+    InsEncodingFmt_rm,
+
+    // "eax, i32" addressing mode
+    InsEncodingFmt_a4,
+
+    // "reg, reg" addressing mode
+    InsEncodingFmt_rr,
+
+    InsEncodingFmt_none,
+    InsEncodingFmt_count = InsEncodingFmt_none
+};
+
+enum InsFlag : unsigned int
```

Comment:
Why is this `unsigned int` instead of `unsigned char`? You'll be wasting 3 bytes per entry.

Reply:
It would need to be `unsigned short` at least. There are a number of switch tables (such as `DstDstSrc`, `DstSrcSrc`, `DstSrcImm`, `TakesRex`, `TakesRexW`, `TakesVex`, etc) which should be moved to be flags.